### PR TITLE
Added url to error message, and fixed opening of browsers for macos a…

### PR DESCRIPTION
**Pull request by VedranBiH**
Added the url to the error message when a browser can't be found on non macOS/windows machines.
Also changed the way macOS and linux open a browser. 
Lastly updated the .gitignore to include some specific settings for the Eclipse IDE

### CODE REVIEW

File: src/main/java/leo/client/Browser.java [hunk 1]
```diff
@@ -14,36 +14,33 @@
 import javax.swing.*;
 import java.lang.reflect.Method;
 
-
 public class Browser {
-    private static final String errMsg = "Error attempting to launch web browser";
```
Comment (kasparsiricenko) [inline src/main/java/leo/client/Browser.java:19]: Wrong identation?

Comment (VedranBiH) [reply src/main/java/leo/client/Browser.java]: Sorry about that looks like I had a formatter on save.

**Comment by lukky513** [comment]
Have you tried it on either macOS or Linux? I'll have a look, too, when I get a chance.

File: src/main/java/leo/client/Browser.java [hunk 2]
```diff
@@ -14,33 +14,34 @@
...
-					throw new Exception(String.format("Could not find web browser.%nURL: %s", url));
-				else
-					runtime.exec("xdg-open " + url);
-			}
-		} catch (Exception e) {
-			JOptionPane.showMessageDialog(null, errMsg + ":\n" + e.getLocalizedMessage());
-		}
-	}
-}
+                if (browser == null)
```
Comment (lukky513) [inline src/main/java/leo/client/Browser.java:38]: I know that we don't have style guide yet (something that I'll try to fix), but for the time being, please add braces to the one-liner `if` constructs. I've been trying to do that whenever I met them :)

Comment (VedranBiH) [reply src/main/java/leo/client/Browser.java]: Done

**Comment by VedranBiH** [comment]
> Have you tried it on either macOS or Linux? I'll have a look, too, when I get a chance.

I tried on my personal Mac intel based not sure if someone has the new M1 chips out there. I can spin up a VM after work today to try it with Linux

File: src/main/java/leo/client/Browser.java [hunk 3]
```diff
@@ -20,30 +20,31 @@ public class Browser {
...
+                else {
+                	runtime.exec("xdg-open " + url);
+                }
             }
         } catch (Exception e) {
             JOptionPane.showMessageDialog(null, errMsg + ":\n" + e.getLocalizedMessage());
         }
     }
-} 
+}
```
Comment (kasparsiricenko) [inline src/main/java/leo/client/Browser.java:50]: Removing newline EOF character?

https://unix.stackexchange.com/questions/18743/whats-the-point-in-adding-a-new-line-to-the-end-of-a-file

Comment (VedranBiH) [reply src/main/java/leo/client/Browser.java]: Not really sure how to fix this.
In my Eclipse IDE it shows as 50 lines and no new line.

I tried loading it VSC and it shows a smaller amount of line but again no EOF.

**Comment by lukky513** [comment]
Looks like an improvement overall. We could merge it as-is, but not close the issue yet, as I think there's still some stuff to do. Let me know if you want to work on this further.

**Comment by VedranBiH** [comment]
> Looks like an improvement overall. We could merge it as-is, but not close the issue yet, as I think there's still some stuff to do. Let me know if you want to work on this further.

Sounds good. For the new stuff is refactoring so its not duplicated across the code base?

**Comment by lukky513** [comment]
> > Looks like an improvement overall. We could merge it as-is, but not close the issue yet, as I think there's still some stuff to do. Let me know if you want to work on this further.
> 
> Sounds good. For the new stuff is refactoring so its not duplicated across the code base?

I can't recall duplication at the moment, but what comes to my mind is:

- extract the code for checking the platform to a separate class, as this might be useful for some other things, e.g. workarounds for the problem of the GUI being slightly shifted in Linux
- instead of just being an 'else' clause, we should have a proper check for Linux; and I think in this case we should try to run `xdg-open`
- trying with few different browsers I would push to an else after that; I guess we can assume that in all likelihood it's going to be some flavor of UNIX that will support `which` command

Possibly there was something more, I'll come back to you if I remember it.

**Comment by lukky513** [comment]
@VedranBiH Please let me know if you want to work on this within this PR, otherwise I'll merge it soon.

**Pull request merged**